contrib/raftexample/kvstore.go

```diff
@@ -77,7 +77,7 @@ func (s *kvstore) readCommits(commitC <-chan *string, errorC <-chan error) {
 			if err := s.recoverFromSnapshot(snapshot.Data); err != nil {
 				log.Panic(err)
 			}
-			continue
+			return
```

Comment:
You are right. I didn't think about the case 2. Let me take a closer look and add a test case for case 2 as well. Thanks.